src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -23,27 +23,23 @@ private partial class DiagnosticIncrementalAnalyzer
         /// <summary>
         /// Return all diagnostics that belong to given project for the given StateSets (analyzers) either from cache or by calculating them
         /// </summary>
-        private async Task<ProjectAnalysisData> GetProjectAnalysisDataAsync(
+        private async Task<ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>> ComputeDiagnosticAnalysisResultsAsync(
             CompilationWithAnalyzersPair? compilationWithAnalyzers, Project project, ImmutableArray<StateSet> stateSets, CancellationToken cancellationToken)
         {
             using (Logger.LogBlock(FunctionId.Diagnostics_ProjectDiagnostic, GetProjectLogMessage, project, stateSets, cancellationToken))
             {
                 try
                 {
                     var version = await GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
-                    var existingData = await ProjectAnalysisData.CreateAsync(project, stateSets, cancellationToken).ConfigureAwait(false);
 
-                    if (existingData.Version == version)
-                        return existingData;
```

Comment:
this would either:

1. not ever succeed (in the case where `ForceProjectAnalysisAsync` had not run).  In which case, it is fine to remove and just do the computation path below.
2. succeed, if it had run and no other changes to the project had been made.  However, the check for that is now pulled into the caller.  So these checks are pointless and this method just bcomes a helper that always computes the results.